File: pset6/mario.py
#-- Mario in Python --
#K.Yap March 2017
#Creates a half pyramid of hashes with the height based off a user's input
#

# ask valid user input (from http://stackoverflow.com/questions/23294658/asking-the-user-for-input-until-they-give-a-valid-response)
while True:
    try:
        height = int(input("Height: "))
    except ValueError:
        print("Please enter a number between 1 and 23")
        continue #try again - return to start of the loop
    
    if (height < 0) or (height > 23):
        print("Please enter a number between 1 and 23")
        continue
    else:
        break #validated, we can now exit loop

#with that out of the way, time to actually print the pyramid
for i in range(height):
    for j in range (height-i-1): #print this number of spaces
        print(" ", end='')
    for k in range (i+2): #print this number of hashes
        print("#", end='')
    print() #go to the next line and repeat the above two loops
    

File: pset1/mario.c
/*
-- Mario --
K.Yap August 2016
Creates a half pyramid of hashes with the height based off a user's input
*/

#include <stdio.h>
#include <cs50.h>

int main(void) 
{
    int height; //declare the variable before the do loop
    do // use a do loop as the check that user input is in the specified range
    {
        printf("Height: ");  //ask for user input
        height = get_int();
    }
    while (height <0 || height >23); //check to make sure input is within 0-23 otherwise ask height again
    
    for (int i = 0; i < height; i++)    
        {
            for (int j = (height-i-1); j > 0; j--) //print this number of spaces
            {
                printf(" ");
            }
            for (int k = (i+2); k > 0; k--) //print this number of hashes
            {
                printf("#");
            }
            printf("\n"); //go to the next line and then repeat the above two loops
        }
    
    }